### ISSUE TITLE
Element 0 of tensors does not require grad and does not have a grad_fn

### ISSUE THREAD

**Issue by sunnyHelen** [nianticlabs/monodepth2#24]
Sorry to bother you. 
I changed some place of the code and got an error "Element 0 of tensors does not require grad and does not have a grad_fn". I think it's here
![image](https://user-images.githubusercontent.com/28101548/59891751-bd262e00-9419-11e9-9d4c-36c21e290d0a.png)
![image](https://user-images.githubusercontent.com/28101548/59891761-c4e5d280-9419-11e9-9cbb-9058b2ffbc40.png)
![image](https://user-images.githubusercontent.com/28101548/59891740-b697b680-9419-11e9-9ab8-3d2f6c1b685b.png)
So, I want to ask you the why we use "with torch.no_grad()" here. If I delete it, will it affect the result?
Thanks a lot for your help~

**Comment by daniyar-niantic** [comment]
Hi @sunnyHelen 

The code with `torch.no_grad()` is used in validation only. The main reason it is there is to not store network activations in GPU memory when doing validation, as we are not training from that set of images. It also may be helpful to catch bugs, such as the one you have posted. It seems that you are trying to backpropagate on the validation set. Without seeing the changes you introduced I'm not sure I can help.

Thanks!

**Comment by mdfirman** [comment]
Yes – if you remove `with torch.no_grad():` here it shouldn't have any effect, as long as you are not trying to update network weights. You can read more [here](https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html#gradients).

Thanks
